### DISCUSSION TITLE
Different loss function scores for generator and discriminator

### DISCUSSION THREAD

**Discussion by KartikeyKansal1** [AllenCellModeling/cyto-dl#350] [General]
Hey, is there a way in cyto-dl to obtain separate loss function scores for the discriminator and the generator?

**Comment by benjijamorris** [comment]
The GAN loss should be split by generator and discriminator (see mlflow screenshot below). Are you thinking for the generator to split out the GAN loss and the reconstruction loss?
![image](https://github.com/AllenCellModeling/cyto-dl/assets/54606172/bc32cc43-c4e5-4ea5-a67a-e24a44afaf24)

**Comment by KartikeyKansal1** [comment]
Hi @benjijamorris, thanks for your reply and sorry about the delay in getting back. Yes I'm looking for this only. Is there a way to print this in the metrics.csv?

**Comment by benjijamorris** [comment]
I think the same metrics get logged regardless of whether you use a csv/tensorboard/mlflow/etc. logger. Running a quick test gave me these columns in my logged csv. Is this different from what you're getting? If so, could you share which columns are logged?
![image](https://github.com/AllenCellModeling/cyto-dl/assets/54606172/37bbb3a4-cb77-4e84-9864-e6778fae3938)

**Comment by KartikeyKansal1** [comment]
Yes I get different log output
<img width="814" alt="image" src="https://github.com/AllenCellModeling/cyto-dl/assets/56447914/87199eb1-9f13-4c57-8e28-5fd6260d5731">

**Comment by benjijamorris** [comment]
Weird! Is this on main running `python cyto_dl/train.py experiment=im2im/gan`?

**Comment by KartikeyKansal1** [comment]
This is im2im/labelfree

**Comment by benjijamorris** [comment]
ah! Labelfree doesn't use a GAN, so you won't see any GAN-related losses being logged. I'm going to close this issue, but if you want to use a GAN, the command I sent above should work very similarly to `im2im/labelfree`.
